### PR TITLE
[release/5.0] Add invocation to NuGet authentication scripts for internal feeds

### PR DESCRIPTION
**Pull request by hoyosjs**
Needed to be able to authenticate internal feeds.

### CODE REVIEW

**Review by mmitche — APPROVED** [review]

**Comment by mmitche** [comment]
Did this get tested on the internal PR? I normally would just plop this change on top of that

**Comment by hoyosjs** [comment]
I had a test PR with a build. This is apparently not enough (it is for the coreclr, mono, and libraries, but not for the installer)

**Comment by hoyosjs** [comment]
Full green build: https://dnceng.visualstudio.com/internal/_git/dotnet-runtime/pullrequest/12290

File: eng/pipelines/common/global-build-job.yml [hunk 1]
```diff
@@ -54,12 +54,33 @@ jobs:
...
+
       - ${{ each variable in parameters.variables }}:
         - ${{ variable }}
 
     steps:
     - template: /eng/pipelines/common/clone-checkout-bundle-step.yml
 
+    - ${{ if ne(variables['System.TeamProject'], 'public') }}:
+      - ${{ if ne(parameters.osGroup, 'Windows_NT') }}:
+        - task: Bash@3
```
Comment (safern) [inline eng/pipelines/common/global-build-job.yml:68]: Could we have done this with nuget authenticate plugin?

Comment (hoyosjs) [reply eng/pipelines/common/global-build-job.yml:68]: I thought so at one point, but Matt recommended for this one (for cross org packages I believe).

Comment (safern) [reply eng/pipelines/common/global-build-job.yml:68]: Ok, just curious as we already use NuGet authenticate for cross org restoring for the icb merge restore.

Comment (hoyosjs) [reply eng/pipelines/common/global-build-job.yml:68]: @mmitche Any reason to prefer one over the other?

**Pull request merged**